src/sentry/eventstream/kafka/backend.py

```diff
@@ -252,14 +254,83 @@ def relay(self, consumer_group, commit_log_topic,
             initial_offset_reset=initial_offset_reset,
         )
 
-        consumer.subscribe([self.publish_topic])
+        owned_partition_offsets = {}
+
+        def on_assign(consumer, partitions):
+            logger.debug('Received partition assignment: %r', partitions)
+
+            for i in partitions:
+                if i.offset is OFFSET_INVALID:
```

Comment:
Dumb question, is using `is` on ints guaranteed or an implementation detail? I know it only works for "small ints" but that's all I remember.

Reply:
Ooooh, good catch, I totally forgot this doesn't work on integers outside of a predefined range on some versions. Will fix.